packages/forms/src/HasForm.php

```diff
@@ -62,6 +63,13 @@ public function getUploadedFileUrl($name, $disk)
 
         if (! $path) return null;
 
+        if ($disk == 's3' && Storage::disk($disk)->getVisibility($path) == 'private') {
```

Comment:
Checking the disk name is a bit dangerous since you could change this to anything you want. Just because you're using `s3` doesn't mean the disk needs to be called `s3` - instead you want to check against the disk's _driver_.